plugins/modules/postgresql_user.py

```diff
@@ -502,21 +508,36 @@ def user_should_we_change_password(current_role_attrs, user, password, encrypted
                 # or we cannot check it properly, e.g. due to missing dependencies
                 pwchanging = True
 
-        # 32: MD5 hashes are represented as a sequence of 32 hexadecimal digits
-        #  3: The size of the 'md5' prefix
         # When the provided password looks like a MD5-hash, value of
         # 'encrypted' is ignored.
-        elif (password.startswith('md5') and len(password) == 32 + 3) or encrypted == 'UNENCRYPTED':
+        elif is_pg_passwd_md5(password) or encrypted == 'UNENCRYPTED':
             if password != current_password:
                 pwchanging = True
         elif encrypted == 'ENCRYPTED':
-            hashed_password = 'md5{0}'.format(md5(to_bytes(password) + to_bytes(user)).hexdigest())
-            if hashed_password != current_password:
+            default_pw_encryption = get_passwd_encryption(cursor)
+
+            if default_pw_encryption == 'md5':
+                hashed_password = 'md5{0}'.format(md5(to_bytes(password) + to_bytes(user)).hexdigest())
+                if hashed_password != current_password:
+                    pwchanging = True
+            elif default_pw_encryption == 'scram-sha-256':
+                # https://github.com/ansible-collections/community.postgresql/issues/688
+                # When the current password is not none and is not
+                # hashed as scram-sha-256 / not explicitly declared as plain text
+                # (if we are here, these conditions should be met)
+                # but the default password encryption is scram-sha-256, update the password.
+                # Can be relevant when migrating from older version of postgres.
```

Comment:
Wouldn't this imply that the password will always be updated if the encryption method is 'scram-sha-256'?